internal/testutil/metrics.go

```diff
@@ -0,0 +1,77 @@
+package testutil
+
+import (
+	"bytes"
+	"context"
+	"strings"
+
+	io_prometheus_client "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+)
+
+func ParseMetrics(metricBytes []byte) (map[string][]*io_prometheus_client.Metric, error) {
+	reader := bytes.NewReader(metricBytes)
+	parser := &expfmt.TextParser{}
+
+	mfs, err := parser.TextToMetricFamilies(reader)
+	if err != nil {
+		return nil, err
+	}
+
+	metrics := make(map[string][]*io_prometheus_client.Metric)
+	pfx := strings.ReplaceAll("package-operator", "-", "_") + "_"
+
+	for name, mf := range mfs {
+		if strings.HasPrefix(name, pfx) {
+			metrics[strings.TrimPrefix(name, pfx)] = mf.GetMetric()
+		}
+	}
+	return metrics, nil
+}
+
+func VerifyMetrics(ctx context.Context, metric, label, value string) (bool, error) {
```

Comment:
proposal for a new name: MetricVectorExists

Needs a comment describing what this function does (specifically that it fetches the metrics first).